File: ios/HandLandmarksFrameProcessor/HandLandmarkerHolder.swift
//
//  HandLandmarkerHolder.swift
//  demo_ring2
//
//  Created by Macos on 30/5/25.
//

import MediaPipeTasksVision

class HandLandmarkerHolder {
  static let shared = HandLandmarkerHolder()
  
  private(set) var handLandmarker: HandLandmarker?
  public var frameWidth: Int = 0;
  public var frameHeight: Int = 0;
  
  private init() {} // Private initializer to enforce singleton pattern
  
  func initializeHandLandmarker(with options: HandLandmarkerOptions) throws {
    self.handLandmarker = try HandLandmarker(options: options)
  }
  
  func clearHandLandmarker() {
    self.handLandmarker = nil
  }
  
  func getHandLandmarker() -> HandLandmarker? {
    return self.handLandmarker
  }
}

File: android/app/src/main/java/com/demo_ring2/HandLandmarkerHolder.kt
package com.demo_ring2

import com.google.mediapipe.tasks.vision.handlandmarker.HandLandmarker

class HandLandmarkerHolder {
    companion object {
        var handLandmarker: HandLandmarker? = null
        var frameWidth = 0f
        var frameHeight = 0f
    }
}
